pkg/rpc/client/client.go

```diff
@@ -119,7 +119,7 @@ func (c *Client) Init() error {
 		return fmt.Errorf("failed to get network magic: %w", err)
 	}
 	c.network = version.Magic
-	c.stateRootInHeader = version.StateRootInHeader
+	c.stateRootInHeader = version.Protocol.StateRootInHeader
```

Comment:
Both of the values should check for `Protocol.X` first and then use values from old locations if there is nothing there.